File: 0049-group-anagrams/0049-group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        groups = {}
        
        for s in strs:
            counts = [0 for i in range(26)]
            for char in s:
                counts[ord(char) - 97] += 1
            
            code = ""
            for i in range(26):
                if counts[i] != 0:
                    code += chr(i + 97)
                    code += str(counts[i])
            if code in groups:
                groups[code].append(s)
            else:
                groups[code] = [s]
        
        res = [value for key, value in groups.items()]
        return res

File: 0049-group-anagrams/0049-group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map<string, vector<string>> map;
        vector<vector<string>> res;
        for (int i = 0; i < strs.size(); i++)
        {
            string curWord = strs[i];
            sort(strs[i].begin(), strs[i].end());
            map[strs[i]].push_back(curWord);
        }
        for (auto list : map)
        {
            res.push_back(list.second);
        }
        return res;
        
    }
};